scripts/apps/archive/controllers/UploadController.ts

```diff
@@ -258,18 +259,23 @@ export function UploadController(
     };
 
     $scope.upload = function() {
-        var promises = [];
+        if (isEmpty($scope.items)) {
+            return Promise.resolve();
+        }
+
+        let promise = Promise.resolve();
 
-        _.each($scope.items, (item) => {
+        // upload items in sequence, and resolve when all are done
+        return Promise.all($scope.items.map((item) => {
```

Comment:
Why do you need `Promise.all` if you want it to be sequential. Wouldn't this work?

```
var p = Promise.resolve();

$scope.items.forEach((item) => {
    p.then(() => uploadFile(item));
});

return p;
```

Reply:
it's just to return list of uploaded files at the end, and not just the last one

Reply:
Could you instead implement and use `promiseAllSerial`? You can check some examples [here](https://decembersoft.com/posts/promises-in-serial-with-array-reduce/).

Wouldn't it be better to fix async upload in the first place? What's wrong with it?

Reply:
there is no bug in our code, but if you upload many images at once sometimes it fails without any obvious reason, post request is rejected before reaching server

Reply:
not sure why do that promiseAllSerial, it's doing basically the same thing, but imo less readable because they use reduce to make a list from a list, instead of using a map..

Reply:
I'm not saying you should implement it as on that website. I'd only want the function to be a standalone function and have the same interface as `Promise.all` meaning it would take an array of promises and when resolved, would return an array with the results of the promises.